firefox-ios/firefox-ios-tests/Tests/ClientTests/BrowserViewControllerTests.swift

```diff
@@ -138,10 +138,10 @@ class BrowserViewControllerTests: XCTestCase, StoreTestUtility {
         subject.tabManager(tabManager, didSelectedTabChange: testTab, previousTab: testTab, isRestoring: false)
         wait(for: [expectation])
 
-        let actionCalled = try XCTUnwrap(mockStore.dispatchedActions[2] as? GeneralBrowserAction)
+        let actionCalled = try XCTUnwrap(mockStore.dispatchedActions[3] as? GeneralBrowserAction)
```

Comment:
had to update these since it seems that even though we are mocking the store, the concrete middleware is being created + responding causing more actions to be dispatched. 

opened a separate ticket to address this.